arrows/core/transform_detected_object_set.cxx

```diff
@@ -0,0 +1,279 @@
+
+  config->merge_config( config_in );
+  this->src_camera_krtd_file_name = config->get_value< std::string > ( "src_camera_krtd_file_name" );
+  this->dest_camera_krtd_file_name = config->get_value< std::string > ( "dest_camera_krtd_file_name" );
+
+  this->src_camera = kwiver::vital::read_krtd_file( this->src_camera_krtd_file_name );
+  this->dest_camera = kwiver::vital::read_krtd_file( this->dest_camera_krtd_file_name );
+}
+
+
+// ------------------------------------------------------------------
+bool
+transform_detected_object_set::
+check_configuration( vital::config_block_sptr config ) const
+{
+  kwiver::vital::config_difference cd( this->get_configuration(), config );
+  const auto key_list = cd.extra_keys();
+
+  if ( ! key_list.empty() )
+  {
+    LOG_WARN( logger(), "Additional parameters found in config block that are not required or desired: "
+              << kwiver::vital::join( key_list, ", " ) );
+  }
+
+  return true;
+}
+
+
+// ------------------------------------------------------------------
+Eigen::Vector3d
+transform_detected_object_set::
+backproject_to_height( const kwiver::vital::camera_perspective_sptr camera,
+		       const Eigen::Vector2d img_pt ) const
+{
+  // Back an image point to a specified world height
+  double height = 0.0;
+
+  Eigen::Vector3d img_pt_3;
+  img_pt_3 << img_pt, 1.0;
+
+  Eigen::Vector3d npt = camera->intrinsics()->as_matrix().colPivHouseholderQr().solve( img_pt_3 );
+  Eigen::Matrix3d M = camera->rotation().matrix();
+  Eigen::Vector3d t = camera->translation();
+
+  M(0,2) = height * M(0,2) + t(0);
+  M(1,2) = height * M(1,2) + t(1);
+  M(2,2) = height * M(2,2) + t(2);
+
+  Eigen::Vector3d wpt = M.colPivHouseholderQr().solve( npt );
+  auto output = Eigen::Vector3d(wpt(0) / wpt(2), wpt(1) / wpt(2), height);
+
+  return output;
+}
+
+// ------------------------------------------------------------------
+Eigen::Vector3d
+transform_detected_object_set::
+backproject_to_plane( const kwiver::vital::camera_perspective_sptr camera,
+		      const Eigen::Vector2d img_pt,
+		      const Eigen::Vector4d plane ) const
+{
+  // Back an image point to a specified world plane
+  Eigen::Vector3d img_pt_3;
+  img_pt_3 << img_pt, 1.0;
+
+  Eigen::Vector3d npt = camera->intrinsics()->as_matrix().colPivHouseholderQr().solve( img_pt_3 );
+  Eigen::Matrix3d M = camera->rotation().matrix().transpose();
+
+  auto n = Eigen::Vector3d(plane(0), plane(1), plane(2));
+  double d = plane(3);
+
+  Eigen::Vector3d Mt = M * camera->translation();
+  Eigen::Vector3d Mp = M * npt;
+
+  return Mp * (n.dot( Mt ) - d) / n.dot( Mp ) - Mt;
+}
+
+// ------------------------------------------------------------------
+Eigen::Matrix<double, 8, 3>
+transform_detected_object_set::
+backproject_bbox( const kwiver::vital::camera_perspective_sptr camera,
+		  const Eigen::Vector4d box ) const
+{
+  // project base of the box to the ground
+  auto base_pt = Eigen::Vector2d((box(0) + box(2)) / 2, box(3));
```

Comment:
FYI, `vector_2d` is a kwiver typedef for `Eigen::Vector2d`.  We tend to use that in kwiver code instead of the Eigen type directly.  Same for 3D.  It's actually `kwiver::vital::vector_2d` we often do `using namespace kwiver::vital;` in the cxx files to shorten the name.